controllers/v1/criteriasController.js

```diff
@@ -934,5 +934,144 @@ module.exports = class Criterias extends Abstract {
     })
   }
 
+  async uploadCriterias(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (!req.files || !req.files.criterias) {
+          throw "Csv file for criterias should be selected"
+        }
+
+        let criteriaData = await csv().fromString(req.files.criterias.data.toString())
+
+        let criteriaDocumentArray = criteriaData.map(criteria => {
+          let criteriaStructure = {
+            owner: req.userDetails.id,
+            name: criteria.criteriaName,
+            description: criteria.criteriaName,
+            resourceType: [
+              "Program",
+              "Framework",
+              "Criteria"
+            ],
+            language: [
+              "English"
+            ],
+            keywords: [
+              "Keyword 1",
+              "Keyword 2"
+            ],
+            concepts: [
+              {
+                identifier: "LPD20100",
+                name: "Teacher_Performance",
+                objectType: "Concept",
+                relation: "associatedTo",
+                description: null,
+                index: null,
+                status: null,
+                depth: null,
+                mimeType: null,
+                visibility: null,
+                compatibilityLevel: null
+              },
+              {
+                identifier: "LPD20400",
+                name: "Instructional_Programme",
+                objectType: "Concept",
+                relation: "associatedTo",
+                description: null,
+                index: null,
+                status: null,
+                depth: null,
+                mimeType: null,
+                visibility: null,
+                compatibilityLevel: null
+              },
+              {
+                identifier: "LPD20200",
+                name: "Teacher_Empowerment",
+                objectType: "Concept",
+                relation: "associatedTo",
+                description: null,
+                index: null,
+                status: null,
+                depth: null,
+                mimeType: null,
+                visibility: null,
+                compatibilityLevel: null
+              }
+            ],
+            createdFor: [
+              "0125747659358699520",
+              "0125748495625912324"
+            ],
+            evidences: [],
+            deleted: false,
+            externalId: criteria.criteriaID,
+            owner: req.userDetails.id,
+            timesUsed: 12,
+            weightage: 20,
+            aremarks: "",
+            name: criteria.criteriaName,
+            description: criteria.criteriaName,
+            criteriaType: "auto",
+            score: "",
+            flag: "",
+            rubric: {
```

Comment:
Please make this dynamic, it should work with how many ever levels are there in CSV. @aman-tunerlabs

Reply:
modified